pylint/checkers/modified_iterating_checker.py

```diff
@@ -90,14 +90,10 @@ def _modified_iterating_check(
         elif self._modified_iterating_set_cond(node, iter_obj):
             msg_id = "modified-iterating-set"
         if msg_id:
-            if isinstance(iter_obj, nodes.Attribute):
-                obj_name = iter_obj.attrname
-            else:
-                obj_name = iter_obj.name
             self.add_message(
                 msg_id,
                 node=node,
-                args=(obj_name,),
+                args=(iter_obj._repr_name(),),
```

Comment:
Since I found this to be useful here, we could consider making this public in astroid 3.0.

Reply:
There's an issue somewhere to make a name/attrname placeholder (it causes like 50% of all our crashes)